datadog_checks_dev/datadog_checks/dev/tooling/utils.py

```diff
@@ -306,3 +310,17 @@ def has_e2e(check):
                     if 'pytest.mark.e2e' in test_file.read():
                         return True
     return False
+
+
+def has_legacy_signature(check):
+    """
+    Validate that the given check does not use the legacy agent signature (contains agentConfig)
+    """
+    for path, _, files in os.walk(get_check_directory(check)):
+        for fn in files:
+            if fn.endswith('.py'):
+                with open(os.path.join(path, fn)) as test_file:
+                    for num, line in enumerate(test_file):
+                        if "__init__" in line and "agentConfig" in line:
+                            return str(fn), num
+    return False
```

Comment:
Since we are returning data, it would be more natural to call the method `find_legacy_signature` for example and return `None` if not found.

Reply:
Sure, I agree.